test/contrib/isotp_soft_socket.uts

```diff
@@ -508,30 +506,32 @@ assert(len(isotp) == 0)
 
 = Check if not sending the first CF will make the socket timeout
 
-with TestSocket(CAN) as isocan, ISOTPSoftSocket(isocan, sid=0x641, did=0x241) as s, TestSocket(CAN) as cans:
+with TestSocket(CAN) as isocan, ISOTPSoftSocket(isocan, tx_id=0x641, rx_id=0x241) as s, TestSocket(CAN) as cans:
     cans.pair(isocan)
     cans.send(CAN(identifier = 0x241, data = dhex("10 11 01 02 03 04 05 06")))
     isotp = s.sniff(timeout=0.1)
 
 assert(len(isotp) == 0)
 
 = Check if not sending the first FC will make the socket timeout
-exception = None
+
+# drain log_stream
+log_stream.getvalue()
+
 isotp = ISOTP(data=dhex("01 02 03 04 05 06 07 08 09 0A"))
 
-with TestSocket(CAN) as isocan, ISOTPSoftSocket(isocan, sid=0x641, did=0x241) as s, TestSocket(CAN) as cans:
+with TestSocket(CAN) as isocan, ISOTPSoftSocket(isocan, tx_id=0x641, rx_id=0x241) as s, TestSocket(CAN) as cans:
     cans.pair(isocan)
-    try:
-        s.send(isotp)
-        assert(False)
-    except Scapy_Exception as ex:
-        exception = ex
+    s.send(isotp)
+    time.sleep(1.3)
```

Comment:
`time.sleep` is bad. If you need this in your test, except in some crazy circumstances, this test needs refactoring :/

Reply:
The refactoring turns the send into a `ObjectPipe.send()` for the main thread. A background thread is checking the objectpipe and handling the send in the background (similar as if the kernel would take care). This unit test checks, if an internal timeout occurs after a first frame is sent, which doesn't get acknowledged by a receiver. In order to wait for this timeout to happen, I need the sleep in this case. This was a design decision, since I didn't wanted to make the `close` function blocking, until all messages in the internal tx queue were handled. 

What do you think about this design decision, would it be more correct, to have the "close" function wait, until the internal TX queue is flushed, or should I just discard all messages in the internal queue and close immediately (as it is right now)?